packages/react-components/react-skeleton/src/components/Skeleton/useSkeletonStyles.ts

```diff
@@ -1,33 +1,121 @@
-import { makeStyles, mergeClasses } from '@griffel/react';
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
 import type { SkeletonSlots, SkeletonState } from './Skeleton.types';
 import type { SlotClassNames } from '@fluentui/react-utilities';
+import { tokens } from '@fluentui/react-theme';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
 
 export const skeletonClassNames: SlotClassNames<SkeletonSlots> = {
   root: 'fui-Skeleton',
-  // TODO: add class names for all slots on SkeletonSlots.
-  // Should be of the form `<slotName>: 'fui-Skeleton__<slotName>`
+};
+
+export const skeletonGroupClassName = 'fui-Skeleton-Group';
+
+const skeletonWaveAnimation = {
+  from: {
+    backgroundPositionX: '300%',
+  },
+  to: {
+    backgroundPositionX: '0%',
+  },
+};
+
+const skeletonWaveAnimationRTL = {
+  from: {
+    backgroundPositionX: '0%',
+  },
+  to: {
+    backgroundPositionX: '300%',
+  },
+};
+
+const skeletonPulseAnimation = {
+  from: {
+    opacity: '1',
+  },
+  to: {
+    opacity: '0.4',
+  },
 };
 
 /**
  * Styles for the root slot
  */
-const useStyles = makeStyles({
+const useRootStyles = makeStyles({
   root: {
-    // TODO Add default styles for the root element
-  },
+    '& [class^= "fui-Skeleton-"]': {
+      position: 'relative',
+      height: 'auto',
+      ...shorthands.overflow('hidden'),
+      ...shorthands.margin('5px'),
+      backgroundImage: `linear-gradient(
+        to right,
+        ${tokens.colorNeutralStencil1} 0%,
+        ${tokens.colorNeutralStencil2} 50%,
+        ${tokens.colorNeutralStencil1} 100%)`,
+      backgroundSize: '300% 100%',
+      backgroundPositionX: 'center',
+      backgroundPositionY: 'center',
+      backgroundAttachment: 'fixed',
+      animationIterationCount: 'infinite',
+      animationDuration: '3s',
+      animationTimingFunction: 'linear',
 
-  // TODO add additional classes for different states and/or slots
+      '@media screen and (forced-colors: active)': {
+        backgroundColor: `WindowText
+        linear-gradient(
+          to right,
+          transparent 0%,
+          Window 50%,
+          transparent 100%)
+        `,
+      },
+    },
+  },
+  wave: {
+    '& [class^= "fui-Skeleton-"]': {
+      animationName: skeletonWaveAnimation,
+    },
+  },
+  waveRtl: {
+    '& [class^= "fui-Skeleton-"]': {
+      animationName: skeletonWaveAnimationRTL,
+    },
+  },
+  pulse: {
+    '& [class^= "fui-Skeleton-"]': {
+      animationName: skeletonPulseAnimation,
+      backgroundColor: tokens.colorNeutralStencil1,
+    },
+  },
+  material: {
+    '& [class^= "fui-Skeleton-"]': {
```

Comment:
nested selectors are an antipattern, and in this case it's not necessary. You can propagate the state flag through context and apply the styles directly in the component

Reply:
Feels like adding context just to pass down styles is adding a lot of complication to avoid a qualified nested selector. The other goal was to allow users to apply classNames directly to their own UI (not needing to use ours). This approach would support that as well.


``` <div class="my-ui fui-Skeleton-my-ui">

Reply:
styles would not be passed through context... it would just be to pass the appearance so that components can apply styles. nested selectors are expensive not only in terms of number of CSS rules generated by Griffel but also can result in lower perf due to how the browser evaluates them 

https://github.com/microsoft/fluentui/blob/master/rfcs/react-components/styles-handbook.md#understanding-selector-complexity
https://github.com/microsoft/fluentui/blob/master/rfcs/react-components/styles-handbook.md#nested-selectors

But it should be quite easy to evaluate which approach would be more performant, just simply spin up a demo and profile the selector stats in edge that @spmonahan did a demo of earlier.

![image](https://user-images.githubusercontent.com/20744592/217193608-97ec3907-0cc3-4eb5-9763-17ea248092f2.png)

Reply:
More details on using Selector Stats: https://learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/evaluate-performance/selector-stats

Happy to chat in person about it if that helps too.

Reply:
I have updated the selector to be less heavy